src/common/fs_manager.cpp

```diff
@@ -341,17 +337,16 @@ bool fs_manager::for_each_dir_node(const std::function<bool(const dir_node &)> &
     return true;
 }
 
-void fs_manager::update_disk_stat(bool check_status_changed)
+void fs_manager::update_disk_stat()
 {
+    zauto_read_lock l(_lock);
```

Comment:
Why did you add a read lock here?

Reply:
It should be a write lock.
Because there is a gap between reset_disk_stat() and the follwoing updating. If a reader get the disks info in the gap, zeros which are incorrect will be returned.